sdk/go/bootstrap.go

```diff
@@ -0,0 +1,187 @@
+type BoostrapConfig struct {
+	ThingID     string    `json:"thing_id,omitempty"`
+	Channels    []string  `json:"channels,omitempty"`
+	ExternalID  string    `json:"external_id,omitempty"`
+	ExternalKey string    `json:"external_key,omitempty"`
+	MFThing     string    `json:"mainflux_id,omitempty"`
+	MFChannels  []Channel `json:"mainflux_channels,omitempty"`
+	MFKey       string    `json:"mainflux_key,omitempty"`
+	Name        string    `json:"name,omitempty"`
+	ClientCert  string    `json:"client_cert,omitempty"`
+	ClientKey   string    `json:"client_key,omitempty"`
+	CACert      string    `json:"ca_cert,omitempty"`
+	Content     string    `json:"content,omitempty"`
+	State       int       `json:"state,omitempty"`
+}
+
+func (sdk mfSDK) AddBootstrap(key string, cfg BoostrapConfig) (string, error) {
+	data, err := json.Marshal(cfg)
+	if err != nil {
+		return "", ErrInvalidArgs
+	}
+
+	url := createURL(sdk.bootstrapURL, sdk.bootstrapPrefix, "configs")
+
+	req, err := http.NewRequest(http.MethodPost, url, bytes.NewReader(data))
+	if err != nil {
+		return "", err
+	}
+
+	resp, err := sdk.sendRequest(req, key, string(CTJSON))
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusCreated && resp.StatusCode != http.StatusOK {
+		if err := encodeError(resp.StatusCode); err != nil {
+			return "", err
+		}
+		return "", ErrFailedCreation
+	}
+
+	id := strings.TrimPrefix(resp.Header.Get("Location"), "/things/configs/")
+	return id, nil
+}
+
+func (sdk mfSDK) ViewBoostrap(key, id string) (BoostrapConfig, error) {
+	endpoint := fmt.Sprintf("%s/%s", "configs", id)
+	url := createURL(sdk.bootstrapURL, sdk.bootstrapPrefix, endpoint)
+
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return BoostrapConfig{}, err
+	}
+
+	resp, err := sdk.sendRequest(req, key, string(CTJSON))
+	if err != nil {
+		return BoostrapConfig{}, err
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return BoostrapConfig{}, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		if err := encodeError(resp.StatusCode); err != nil {
+			return BoostrapConfig{}, err
+		}
+		return BoostrapConfig{}, ErrFetchFailed
+	}
+
+	var bc BoostrapConfig
+	if err := json.Unmarshal(body, &bc); err != nil {
+		return BoostrapConfig{}, err
+	}
+
+	return bc, nil
+}
+
+func (sdk mfSDK) UpdateBoostrap(key string, cfg BoostrapConfig) error {
+	data, err := json.Marshal(cfg)
+	if err != nil {
+		return ErrInvalidArgs
+	}
+
+	endpoint := fmt.Sprintf("%s/%s", "configs", cfg.MFThing)
+	url := createURL(sdk.bootstrapURL, sdk.bootstrapPrefix, endpoint)
+
+	req, err := http.NewRequest(http.MethodPut, url, bytes.NewReader(data))
+	if err != nil {
+		return err
+	}
+
+	resp, err := sdk.sendRequest(req, key, string(CTJSON))
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		if err := encodeError(resp.StatusCode); err != nil {
+			return err
+		}
+		return ErrFailedUpdate
```

Comment:
wrap more info into the error returned here

Reply:
It's one or the other, I can't wrap